### PR TITLE
Set up structure for querying GraphQL

### PR DESCRIPTION
**Pull request by tobinus**
The following has been done:
* Use React state and properties in `App` to propagate changes and callbacks to components lower in the hierarchy.
  * Existing GraphQL clients polyfill the promise and fetch API, which are already polyfilled by React. I decided to roll our own solution for better comprehension and simplicity. Most issues I've had (CORS and CSRF) wouldn't have been fixed by external libraries either.
* Create function which encapsulates the GraphQL API. It will send JWT, deal with CORS and CSRF and use the correct URL.
* Add configuration options through environment variables.
* Give up on maintaining the single React test.
* Expand the documentation.

I recommend reading `README.md`.

### CODE REVIEW

**Comment by tobinus** [comment]
I consider myself done with this branch now, so please review and merge.

File: src/App.js [hunk 1]
```diff
@@ -2,15 +2,200 @@ import React, { Component } from 'react';
...
+   * Remove the given productID from the cart.
+   *
+   * Available through props.cart.get('remove').
+   *
+   * If the item is not found in the cart, no error is thrown.
+   * @param productID ID of product to remove from cart.
+   */
+  removeProductFromCart(productID) {
+    this.setState((prevState, props) => {
+      const productIndex = prevState.products.indexOf(productID);
```
Comment (tutturen) [inline src/App.js:173]: Can we use `.includes()` here?

File: src/Main/index.js [hunk 2]
```diff
@@ -9,14 +9,14 @@ import CheckoutPage from '../CheckoutPage';
...
-      <Route path="/signup" component={SignUpPage} />
-      <Route path="/login" component={LogInPage} />
-      <Route path="/cart" component={CartPage} />
-      <Route path="/checkout" component={CheckoutPage} />
+      <Route exact path="/" render={(routeProps) => <ProductPage {...routeProps} {...props}/>} />
+      <Route path="/signup" render={(routeProps) => <SignUpPage {...routeProps} {...props}/>} />
+      <Route path="/login" render={(routeProps) => <LogInPage {...routeProps} {...props}/>} />
+      <Route path="/cart" render={(routeProps) => <CartPage {...routeProps} {...props}/>} />
+      <Route path="/checkout" render={(routeProps) => <CheckoutPage {...routeProps} {...props}/>} />
     </Switch>
```
Comment (tutturen) [inline src/Main/index.js:20]: This is not ideal. We should find a better solution to this in the future.

**Review by tutturen — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (tobinus) [reply src/App.js:173]: .includes() does not return the index where the ID is found?

Comment (tobinus) [reply src/App.js:173]: I'll change to more readable API, but we need the index if we want to remove it.

Comment (tutturen) [reply src/App.js:173]: That is a good point.

**Review by tutturen — APPROVED** [review]

**Pull request merged**